### PR TITLE
Feature/repeat func

### LINKED ISSUE #11: Feature - Create operator that takes a function.
The idea would be to have an operator that takes a function and a count to generate a new list.  

> At first, I thought that was what repeat was doing.

**Example**
```typescript
const generateDot = () => {
  const x = Math.random() * this.width;
  const y = Math.random() * this.height;
  return { x, y };
}

const dots = generate(generateDot, 50); // I couldn't find a name for the operator.
```

### PR DESCRIPTION
**Pull request by MonsieurMan**
Fixes #11.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -348,6 +348,23 @@ repeat(1, 7); //=> list(1, 1, 1, 1, 1, 1, 1)
 repeat("foo", 3); //=> list("foo", "foo", "foo")
 ```
 
+### `times`
+
+Returns a list of given length that contains the value of the given function called with current index.
+
+**Complecity**: `O(n)`
```
Comment (paldepind) [inline README.md]: ~~Complecity~~ Complexity

File: src/index.ts [hunk 2]
```diff
@@ -656,6 +656,22 @@ export function repeat<A>(value: A, times: number): List<A> {
...
+ * @param func Function used to generate list values.
+ * @param times Number of values to generate.
+ */
+export function times<A>(func: (index: number) => A, times: number): List<A> {
+  let l = empty();
+  let i = 0;
+  while(i < times) {
+    l = append(func(i), l);
+    i++;
+  }
```
Comment (paldepind) [inline src/index.ts]: What about using a `for`-loop here?

**Comment by paldepind** [comment]
Really nice. Thank you!

**Comment by paldepind** [comment]
I've just published a new version that includes `times`. Thanks a lot @MonsieurMan 😄

**Comment by MonsieurMan** [comment]
@paldepind It's always a pleasure to contribute to open source packages. :smile:

**Pull request merged**